### PR TITLE
Flowhash livedev 5270 v1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5270

Describe changes:
- flow: optionally use livedev for hash

Should it be optional ? Ie does suricata support having 2 interfaces monitoring the same network (and flows) ?

Should be rebased after #7384

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ips_afp_suri_cfg.
  

Pipeline 7321

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on tlpw1_files_sha256.
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk | | |
| .app_layer.error.ftp-data.parser | 0 | 395 | 0.0% | 

Pipeline 7347

File: src/flow-hash.c [hunk 1]
```diff
@@ -373,28 +389,23 @@ static inline int FlowCompareICMPv4(Flow *f, const Packet *p)
...
-                f->vlan_id[0] == p->vlan_id[0] &&
-                f->vlan_id[1] == p->vlan_id[1])
-        {
+        } else if ((f->dst.addr_data32[0] == IPV4_GET_RAW_IPSRC_U32(ICMPV4_GET_EMB_IPV4(p))) &&
+                   (f->src.addr_data32[0] == IPV4_GET_RAW_IPDST_U32(ICMPV4_GET_EMB_IPV4(p))) &&
+                   f->dp == p->icmpv4vars.emb_sport && f->sp == p->icmpv4vars.emb_dport &&
+                   f->proto == ICMPV4_GET_EMB_PROTO(p) &&
+                   f->recursion_level == p->recursion_level && f->vlan_id[0] == p->vlan_id[0] &&
+                   f->vlan_id[1] == p->vlan_id[1] &&
+                   (f->livedev == p->livedev || g_livedev_mask == 0)) {
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/flow-hash.c:408]: > Why dont we use CmpVlanIds ?

Already proposed in a separate PR cf https://github.com/OISF/suricata/pull/7384

File: src/flow-hash.c [hunk 2]
```diff
@@ -313,10 +331,9 @@ static inline bool CmpFlowPacket(const Flow *f, const Packet *p)
     const uint32_t *f_dst = f->dst.address.address_un_data32;
     const uint32_t *p_src = p->src.address.address_un_data32;
     const uint32_t *p_dst = p->dst.address.address_un_data32;
-    return CmpAddrsAndPorts(f_src, f_dst, f->sp, f->dp, p_src, p_dst, p->sp,
-                            p->dp) && f->proto == p->proto &&
-            f->recursion_level == p->recursion_level &&
-            CmpVlanIds(f->vlan_id, p->vlan_id);
+    return CmpAddrsAndPorts(f_src, f_dst, f->sp, f->dp, p_src, p_dst, p->sp, p->dp) &&
+           f->proto == p->proto && f->recursion_level == p->recursion_level &&
+           CmpVlanIds(f->vlan_id, p->vlan_id) && (f->livedev == p->livedev || g_livedev_mask == 0);
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/flow-hash.c:336]: It is shorter if we have access to the pointer information

File: src/flow-hash.c [hunk 3]
```diff
@@ -305,6 +318,11 @@ static inline bool CmpVlanIds(const uint16_t vlan_id1[2], const uint16_t vlan_id
            ((vlan_id1[1] ^ vlan_id2[1]) & g_vlan_mask) == 0;
 }
 
+static inline bool CmpLiveDevIds(const LiveDevice *livedev, const uint16_t id)
+{
+    return livedev && ((livedev->id ^ id) & g_livedev_mask) == 0;
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/flow-hash.c:323]: Yes, or you can see it the other way around : we don't care about ids if `g_livedev_mask` is 0

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on tlpw1_files_sha256.
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk | | |
| .app_layer.error.ftp-data.parser | 0 | 395 | 0.0% | 

Pipeline 7347
WARNING: THERE IS A KNOWN BAD BASELINE WITH PACKET DROPS. bE MINDFUL OF ANY RESULTS.

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7950

**Pull request closed without merging**